gsmphys/sfc_nst.f

```diff
@@ -561,6 +562,11 @@ subroutine sfc_nst                                                &
           tem     = 1.0 / rho_a(i)
           hflx(i) = hflx(i) * tem * cpinv
           evap(i) = evap(i) * tem * hvapi
+          if (evap(i) .lt. -maxevap(i)) then
+            chh(i) = -maxevap(i)/evap(i)*chh(i)
+            hflx(i) = -maxevap(i)/hflx(i)*chh(i)
```

Comment:
Checking order of operations is supposed to be `(-maxevap/evap) * _var_` and you want to negate it.